### PR TITLE
SS14-12462 Nerf food and drink vending machines

### PR DESCRIPTION
**Pull request by FairlySadPanda**
## About the PR
Reduces the amount of food in all food and vending machines EXCEPT the Honk meal one by about half.

## Why / Balance
Vending machines provide too much food (and drink) at the moment to the crew, robbing the chef/bartender of a reason to exist, and robbing the janitor of a reason to want to refill vending machines early in the round.

## Technical details
This PR adds a new "initialStockQuality" field to vending machines and sets it at 0.33 for almost all food and drink vendors. The intent at the moment is to drop food and drink vending machine stocks by somewhere around a half - two thirds of the time, about two-thirds of the stock of a given item will be missing.

This number can be tuned to discourage people relying on vending machines and make round start a bit more variable when hunting noms.

## Media

https://github.com/space-wizards/space-station-14/assets/732532/29139d1a-98e4-4b64-9670-cd2ef284d1b1

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
:cl:
- tweak: Food and drink stocks in vending machines has been reduced to encourage people to use the kitchen and bar.

### CODE REVIEW

**Comment by FairlySadPanda** [comment]
Have moved this ticket to cover #26000 rather than the general Chef refactor @metalgearsloth proposed in #17183. As it ties into that feature, and I can't be fucked to change branch name, the branch name will remain labelled by the feature.

**Comment by VektorZ1** [comment]
25999 :trollface:

**Comment by FairlySadPanda** [comment]
Have moved to #12462

**Comment by FairlySadPanda** [comment]
I have been informed that the Honk vending machine is just for happy meal boxes and toys

I thought they stocked literal happy meals complete with food. C'est la vie.

**Comment by UbaserB** [comment]
can you make restocks cheaper as well

**Comment by FairlySadPanda** [comment]
> can you make restocks cheaper as well

Out of scope for this PR, sorry

> This fits more in the `BasicRoundstartVariation` IMO.

Maybe, will wait for review as this greasy code addition might well be fine

> _<nitpick mode please ignore>Shouldn't the field be called `Initial Stock **Quantity**` rather than `Quality` 🤔?</nitpick mode please ignore>_

Fair question - quantity is "how many of this thing are in this thing" and quality is "how good is this thing". "How good is the stock of this vending machine?" Because we can express that as a concept between "completely bad" - no stock at all - and "completely good" - perfect stock levels, we can express the progression the vending machine is on as any positive rational number between 0 and 1, with overall stock quality approaching completely good as the number approaches 1.

**Comment by Tayrtahn** [comment]
InitialStockFullness?

**Comment by potato1234x** [comment]
it should have a lower chance to remove an item when there is more of them, the gibb machines are meant to have more dr gibb than other drinks and making them sometimes have none kinda sucks

**Comment by musicmanvr** [comment]
the machines are always empty anyways and typically never restocked, this is un-needed and a bad change

**Comment by Sirionaut** [comment]
While I like the change in general, I do wanna point out that maps like Packed only have 3 food vending machines if I recall correctly.

**Comment by FairlySadPanda** [comment]
> While I like the change in general, I do wanna point out that maps like Packed only have 3 food vending machines if I recall correctly.

Yeah. Having it as a prototype-level value means that a mapper should be able to increase quality for specific vending machines. The 0.33 value is the default atm.

File: Content.Shared/VendingMachines/VendingMachineComponent.cs [hunk 1]
```diff
@@ -123,6 +123,13 @@ public sealed partial class VendingMachineComponent : Component
         public float DenyAccumulator = 0f;
         public float DispenseOnHitAccumulator = 0f;
 
+        /// <summary>
+        /// The quality of the stock in the vending machine on spawn.
+        /// Represents the percentage chance (0.0f = 0%, 1.0f = 100%) each set of items in the machine is fully-stocked.
+        /// If not fully stocked, the stock will have a random value between 0 (inclusive) and max stock (exclusive).
+        /// </summary>
+        public float InitialStockQuality = 1.0f;
```
Comment (EmoGarbage404) [inline Content.Shared/VendingMachines/VendingMachineComponent.cs:131]:
```suggestion
        [DataField]
        public float InitialStockQuality = 1.0f;
```

**Pull request merged**